bundles/org.openhab.binding.venstarthermostat/src/main/java/org/openhab/binding/venstarthermostat/internal/handler/VenstarThermostatHandler.java

```diff
@@ -382,13 +382,11 @@ private void setFanMode(VenstarFanMode fanmode) {
     }
 
     private void setAwayMode(VenstarAwayMode away) {
-        VenstarScheduleMode schedule = infoData.getScheduleMode();
-        updateSettings(away, schedule);
+        updateAwayMode(away);
```

Comment:
Rather then call updateAwayMode, just put the logic from it in this function, no need to have 2